### PR TITLE
Added Optional User Authentification When Running yarn prod

### PR DESCRIPTION
**Pull request by hunter-bennett28**
Added optional log in lock.

Trimmed a lot of unnecessary code out of App.js

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -29,3 +29,22 @@ Body {
...
+  padding: 5%;
+  border-radius: 30px;
+  background-color: #444;
+}
+
+.loginQuestionText {
+  color: #aaa;
+}
+
+.linkText {
```
Comment (conninator2000) [inline src/App.css:46]: I like it, the poor mans <a>

File: src/components/LoginComponent.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  );
+
+  const handleRegisterButton = async () => {
+    try {
+      const result = await (state.login
+        ? signInUser(state.email, state.pass)
+        : registerUser(state.email, state.pass));
+      notifyLogIn();
+      sessionStorage.setItem('user', JSON.stringify(result));
+      history.push('/home');
```
Comment (conninator2000) [inline src/components/LoginComponent.js:62]: Just curious as to what this does

Comment (conninator2000) [reply src/components/LoginComponent.js:62]: Id assume push them to the homepage once they have logged in?

File: src/components/ProductBacklogListComponent.js [hunk 3]
```diff
@@ -330,24 +333,33 @@ const ProductBacklogListComponent = ({ refreshProjects, displayPopup }) => {
...
   };
 
+  // Only allow access if logged in
+  if (
+    process.env.REACT_APP_USE_AUTH &&
+    !loggedIn &&
+    !sessionStorage.getItem('user')
+  ) {
+    console.log('no user found');
+    return <Redirect to='/login' />;
```
Comment (conninator2000) [inline src/components/ProductBacklogListComponent.js:370]: I like it, we will have to make sure that it is present on all components that we make going forward

**Review by conninator2000 — APPROVED** [review]
It looks good, I like the additions to each of the files to make sure it is authorized.

File: src/components/LoginComponent.js (hunk 2 above)
Comment (hunter-bennett28) [reply src/components/LoginComponent.js:62]: yeah it's a weirdly named way of redirecting to home

**Pull request merged**